solr_manager/docker-compose.yml

```diff
@@ -0,0 +1,145 @@
+        ENV: ${ENV:-dev}
+        POETRY_VERSION: ${POETRY_VERSION:-1.5.1}
+        SOLR_PASSWORD: ${SOLR_PASSWORD:-solr}
+        SOLR_USER: ${SOLR_USER:-solrRocks}
+        ZK_HOST: ${ZK_HOST:-zoo1:2181,zoo2:2181,zoo3:2181}
+    env_file:
+      - .env
+    volumes:
+      - .:/app
+    stdin_open: true
+    depends_on:
+      - zoo1
+      - zoo2
+      - zoo3
+      - solr1
+      - solr2
+      - solr3
+      - cluster_security
+    tty: true
+    container_name: solr_manager
+    networks:
+      - solr
+  solr1:
+    build:
+      context: ..
+      dockerfile: solr8.11.2_files/Dockerfile
+      target: embedded_zookeeper
+    container_name: solr1
+    ports:
+     - "8981:8983"
+    environment:
+      - ZK_HOST=zoo1:2181,zoo2:2181,zoo3:2181
+    networks:
+      - solr
+    depends_on:
+      - zoo1
+      - zoo2
+      - zoo3
+  solr2:
+    build:
+      context: ..
+      dockerfile: solr8.11.2_files/Dockerfile
+      target: embedded_zookeeper
+    container_name: solr2
+    ports:
+     - "8982:8983"
+    environment:
+      - ZK_HOST=zoo1:2181,zoo2:2181,zoo3:2181
+    networks:
+      - solr
+    depends_on:
+      - zoo1
+      - zoo2
+      - zoo3
+  solr3:
+    build:
+      context: ..
+      dockerfile: solr8.11.2_files/Dockerfile
+      target: embedded_zookeeper
+    container_name: solr3
+    ports:
+     - "8983:8983"
+    environment:
+      - ZK_HOST=zoo1:2181,zoo2:2181,zoo3:2181
+    networks:
+      - solr
+    depends_on:
+      - zoo1
+      - zoo2
+      - zoo3
+  zoo1:
+    image: zookeeper:3.6.2
+    container_name: zoo1
+    restart: always
+    hostname: zoo1
+    ports:
+      - 2181:2181
+      - 7001:7000
+    environment:
+      ZOO_MY_ID: 1
+      ZOO_SERVERS: server.1=zoo1:2888:3888;2181 server.2=zoo2:2888:3888;2181 server.3=zoo3:2888:3888;2181
+      ZOO_4LW_COMMANDS_WHITELIST: mntr, conf, ruok
+      ZOO_CFG_EXTRA: "metricsProvider.className=org.apache.zookeeper.metrics.prometheus.PrometheusMetricsProvider metricsProvider.httpPort=7000 metricsProvider.exportJvmInfo=true"
+    networks:
+      - solr
+  zoo2:
+    image: zookeeper:3.6.2
+    container_name: zoo2
+    restart: always
+    hostname: zoo2
+    ports:
+      - 2182:2181
+      - 7002:7000
+    environment:
+      ZOO_MY_ID: 2
+      ZOO_SERVERS: server.1=zoo1:2888:3888;2181 server.2=zoo2:2888:3888;2181 server.3=zoo3:2888:3888;2181
+      ZOO_4LW_COMMANDS_WHITELIST: mntr, conf, ruok
+      ZOO_CFG_EXTRA: "metricsProvider.className=org.apache.zookeeper.metrics.prometheus.PrometheusMetricsProvider metricsProvider.httpPort=7000 metricsProvider.exportJvmInfo=true"
+    networks:
+      - solr
+  zoo3:
+    image: zookeeper:3.6.2
+    container_name: zoo3
+    restart: always
+    hostname: zoo3
+    ports:
+      - 2183:2181
+      - 7003:7000
+    environment:
+      ZOO_MY_ID: 3
+      ZOO_SERVERS: server.1=zoo1:2888:3888;2181 server.2=zoo2:2888:3888;2181 server.3=zoo3:2888:3888;2181
+      ZOO_4LW_COMMANDS_WHITELIST: mntr, conf, ruok
```

Comment:
Would be useful to explain why we need to include this environment variable (without it do commands fail?)

Reply:
I have created this [page](https://hathitrust.atlassian.net/wiki/spaces/HAT/pages/3190292502/Solr+cloud+and+external+Zookeeper+parameters) documenting Solr and Zookeeper setup. I also have added this documentation in the Readme.md of the repository and as comments in the docker-compose.yml.

In the future when we have a clear understanding of our Solr cloud architecture I think we should remove the comments form the docker-compose file. However, I think is fine to have this information closer right now to easily find it, discuss it, and learn about Solr Cloud.

If we do not include these environment variables, the Zokeeper server won't fail, however, is a good practice to include them to define what are the metrics and health checks running in the Zookeeper server. This also increases the security of the server because we can have control of the things running there.